### PR TITLE
"starts with" search; API URL settings

### PR DESCRIPTION
**Pull request by jtrein**
* edits the API URL
* implements the "starts with" search
* some styles changes
* spinny wheel guy

### CODE REVIEW

**Comment by jtrein** [comment]
Done!

File: src/js/components/SearchBar.jsx [hunk 1]
```diff
@@ -49,57 +51,130 @@ const _sameSearchTerm = (value, old) => value.trim() === old.trim();
...
-  handleSearch(e, value) {
+  componentDidUpdate(previousProps, previousState) {
+    // re-run search?
+    if (previousState.startsWithSearch !== this.state.startsWithSearch) {
+      this.handleSearch(null, this.state.value, true);
+    }
+  }
+
+  buildApiUrl(state = this.state) {
+    const baseSearchUrl = `${SEARCH_URL}${state.value.toLowerCase()}`;
```
Comment (Chris-Petty) [inline src/js/components/SearchBar.jsx]: 'value' is quite an uninformative name for a state property, so much so I have no context to suggest a better name.

Comment (Chris-Petty) [inline src/js/components/SearchBar.jsx]: from the docblock of handleSearch describing 'value', searchTerm or currentSearch would be a better name for it in state. Assuming it's talking about the same thing.

File: src/js/components/SearchBar.jsx [hunk 2]
```diff
@@ -49,57 +51,130 @@ const _sameSearchTerm = (value, old) => value.trim() === old.trim();
...
+              borderBottomColor: 'rgba(242, 101, 50, 0.55)',
+              borderTopColor: 'rgba(242, 101, 50, 0.55)',
+              borderLeftColor: 'rgba(242, 101, 50, 0.55)',
+              borderRightColor: 'rgba(242, 101, 50, 0.55)',
+            }}
+            underlineStyle={{
+              borderBottomColor: '#DDD',
+            }}
+          />
+          {/* progress indicator */}
```
Comment (Chris-Petty) [inline src/js/components/SearchBar.jsx]: w00t!

File: src/js/components/SearchBar.jsx [hunk 3]
```diff
@@ -49,57 +51,130 @@ const _sameSearchTerm = (value, old) => value.trim() === old.trim();
...
-        />
-        {/* search icon */}
-        <SearchIcon
-          alt="search"
-          className="icon-search"
-          color={'#CCC'}
+          className="search__option"
+          label="Name or code begins with"
+          labelStyle={{ color: '#777', marginLeft: '-12px' }}
+          onCheck={(e, isChecked) => this.onChangeUserSearchSetting(isChecked)}
```
Comment (Chris-Petty) [inline src/js/components/SearchBar.jsx]: Having difficulty finding this prop for SearchIcon.

**Comment by jtrein** [comment]
ok, code updated per review.

**Pull request merged**